### PR TITLE
[Task] Optimized `SearchController::findAction`

### PR DESCRIPTION
**Pull request by Corepex**

### CODE REVIEW

File: bundles/AdminBundle/Controller/Searchadmin/SearchController.php [hunk 1]
```diff
@@ -77,9 +77,9 @@ public function findAction(Request $request, EventDispatcherInterface $eventDisp
 
         $query = $this->filterQueryParam($allParams['query'] ?? '');
 
-        $types = explode(',', $allParams['type'] ?? '');
-        $subtypes = explode(',', $allParams['subtype'] ?? '');
-        $classnames = explode(',', $allParams['class'] ?? '');
+        $types = explode(',', preg_replace("/[^a-zA-z,]/", '', $allParams['type'] ?? ''));
+        $subtypes = explode(',', preg_replace("/[^a-zA-z,]/", '', $allParams['subtype'] ?? ''));
+        $classnames = explode(',', preg_replace("/[^a-zA-z,]/", '', $allParams['class'] ?? ''));
```
Comment (jdreesen) [inline bundles/AdminBundle/Controller/Searchadmin/SearchController.php]:
```suggestion
        $types = explode(',', preg_replace("/[^a-zA-Z,]/", '', $allParams['type'] ?? ''));
        $subtypes = explode(',', preg_replace("/[^a-zA-Z,]/", '', $allParams['subtype'] ?? ''));
        $classnames = explode(',', preg_replace("/[^a-zA-Z,]/", '', $allParams['class'] ?? ''));
```

Comment (Corepex) [reply bundles/AdminBundle/Controller/Searchadmin/SearchController.php]: thanks :)

Comment (jdreesen) [reply bundles/AdminBundle/Controller/Searchadmin/SearchController.php]: Or maybe:

```suggestion
        $types = explode(',', preg_replace("/[^a-z,]/i", '', $allParams['type'] ?? ''));
        $subtypes = explode(',', preg_replace("/[^a-z,]/i", '', $allParams['subtype'] ?? ''));
        $classnames = explode(',', preg_replace("/[^a-z,]/i", '', $allParams['class'] ?? ''));
```

File: bundles/AdminBundle/Controller/Searchadmin/SearchController.php [hunk 2]
```diff
@@ -77,9 +77,9 @@ public function findAction(Request $request, EventDispatcherInterface $eventDisp
 
         $query = $this->filterQueryParam($allParams['query'] ?? '');
 
-        $types = explode(',', $allParams['type'] ?? '');
-        $subtypes = explode(',', $allParams['subtype'] ?? '');
-        $classnames = explode(',', $allParams['class'] ?? '');
+        $types = explode(',', preg_replace("/[^a-zA-Z,]/", '', $allParams['type'] ?? ''));
+        $subtypes = explode(',', preg_replace("/[^a-zA-Z,]/", '', $allParams['subtype'] ?? ''));
+        $classnames = explode(',', preg_replace("/[^a-zA-Z,]/", '', $allParams['class'] ?? ''));
```
Comment (jdreesen) [inline bundles/AdminBundle/Controller/Searchadmin/SearchController.php]: Or maybe:

```suggestion
        $types = explode(',', preg_replace("/[^a-z,]/i", '', $allParams['type'] ?? ''));
        $subtypes = explode(',', preg_replace("/[^a-z,]/i", '', $allParams['subtype'] ?? ''));
        $classnames = explode(',', preg_replace("/[^a-z,]/i", '', $allParams['class'] ?? ''));
```

File: bundles/AdminBundle/Controller/Searchadmin/SearchController.php (hunk 1 above)
Comment (Corepex) [reply bundles/AdminBundle/Controller/Searchadmin/SearchController.php]: `/i` simplifies the regex - I think that would be the best solution - thanks (again) :)

File: bundles/AdminBundle/Controller/Searchadmin/SearchController.php [hunk 3]
```diff
@@ -77,9 +77,9 @@ public function findAction(Request $request, EventDispatcherInterface $eventDisp
 
         $query = $this->filterQueryParam($allParams['query'] ?? '');
 
-        $types = explode(',', $allParams['type'] ?? '');
-        $subtypes = explode(',', $allParams['subtype'] ?? '');
-        $classnames = explode(',', $allParams['class'] ?? '');
+        $types = explode(',', preg_replace("/[^a-z,]/i", '', $allParams['type'] ?? ''));
+        $subtypes = explode(',', preg_replace("/[^a-z,]/i", '', $allParams['subtype'] ?? ''));
+        $classnames = explode(',', preg_replace("/[^a-z,]/i", '', $allParams['class'] ?? ''));
```
Comment (brusch) [inline bundles/AdminBundle/Controller/Searchadmin/SearchController.php]:
```suggestion
        $classnames = explode(',', preg_replace("/[^a-z0-9_,]/i", '', $allParams['class'] ?? ''));
```
See https://github.com/pimcore/pimcore/blob/528335a4385604a2bf56a168d8b289ab1ee998ab/models/DataObject/ClassDefinition.php#L354 for valid names 4 classes 😊

**Review by brusch — CHANGES_REQUESTED** [review]

File: bundles/AdminBundle/Controller/Searchadmin/SearchController.php (hunk 3 above)
Comment (Corepex) [reply bundles/AdminBundle/Controller/Searchadmin/SearchController.php]: thanks for the hint :)

**Pull request merged**